commonui/src/main/java/com/hyperwallet/android/ui/common/insight/HyperwalletInsight.java

```diff
@@ -18,34 +18,202 @@
 package com.hyperwallet.android.ui.common.insight;
 
 import android.content.Context;
+import android.os.Handler;
 
 import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
 
+import com.hyperwallet.android.Configuration;
+import com.hyperwallet.android.Hyperwallet;
+import com.hyperwallet.android.exception.HyperwalletException;
+import com.hyperwallet.android.insight.collect.ErrorInfo;
+import com.hyperwallet.android.insight.collect.Insight;
+import com.hyperwallet.android.listener.HyperwalletListener;
+
+import java.text.MessageFormat;
 import java.util.Map;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
 
 /**
  * Used for gathering the data necessary for the Insights analytics.
  */
 public class HyperwalletInsight {
     private static HyperwalletInsight sHyperwalletInsight;
+    private Executor mExecutor = Executors.newSingleThreadExecutor();
 
     private HyperwalletInsight() {
     }
 
+    /**
+     * Returns instance of  HyperwalletInsight or initializes it if it is null.
+     *
+     * @return singleton instance of HyperwalletInsight
+     */
     public static synchronized HyperwalletInsight getInstance() {
         if (sHyperwalletInsight == null) {
             sHyperwalletInsight = new HyperwalletInsight();
         }
         return sHyperwalletInsight;
     }
 
-    public void trackClick(@NonNull final Context context, @NonNull final String pageName,
-            @NonNull final String pageGroup, String link, Map<String, String> params) {
-
+    /**
+     * Initializes the Insight library using the given parameters.
+     *
+     * @param context      the context using Insight
+     * @param environment  the application's environment, ie. QA, UAT, PROD
+     * @param programToken the user's program token
+     * @param sdkVersion   the application's Hyperwallet SDK version
+     * @param apiUrl       the URL of the Insight server to send the data to
+     * @param userToken    the user's payee token
+     */
+    public void initializeInsight(final Context context, final String environment, final String programToken,
```

Comment:
we can change the signature to `public void initializeInsight(@NonNull final Context context, @NonNull final Configuration configuration)` and then do all stuff from method `onGetConfigurationSuccess` in here and then initialize Insight